tests/src/test/java/alluxio/underfs/UnderStorageSystemInterfaceIntegrationTest.java

```diff
@@ -126,7 +126,19 @@ public void deleteLargeDirectory() throws IOException {
 
     String[] children = config.getChildren();
     for (String child : children) {
-      Assert.assertFalse(mUfs.exists(child));
+      // Retry for some time to allow list operation eventual consistency for S3 and GCS.
+      // See http://docs.aws.amazon.com/AmazonS3/latest/dev/Introduction.html and
+      // https://cloud.google.com/storage/docs/consistency for more details.
+      // Note: not using CommonUtils.waitFor here because we intend to sleep with a longer interval.
+      boolean childDeleted = false;
+      for (int i = 0; i < 20; i++) {
+        childDeleted = !mUfs.exists(child);
+        if (childDeleted) {
+          break;
+        }
+        CommonUtils.sleepMs(500);
+      }
+      Assert.assertTrue(childDeleted);
```

Comment:
We shouldn't need this retry logic here, since we are calling `exists` directly on each object instead of calling `list`?